wanda/counting_writer.go

```diff
@@ -0,0 +1,21 @@
+package wanda
+
+import (
+	"io"
+)
+
+type countingWriter struct {
+	w io.Writer
+	n int64
+}
+
+func newCountingWriter(w io.Writer) *countingWriter {
+	return &countingWriter{w: w}
+}
+
+// Write writes to the underlying writer and counts the number of bytes written.
```

Comment:
nit: write /writes/data ?

Reply:
this is a golang thing. the comment always looks like `<Function name> does something`